python/ql/src/experimental/semmle/python/Concepts.qll

```diff
@@ -404,3 +404,77 @@ class JWTDecoding extends DataFlow::Node instanceof JWTDecoding::Range {
    */
   predicate verifiesSignature() { super.verifiesSignature() }
 }
+
+/** Provides classes for modeling Email APIs. */
+module EmailSender {
+  /**
+   * A data-flow node that sends an email.
+   *
+   * Extend this class to model new APIs. If you want to refine existing API models,
+   * extend `EmailSender` instead.
+   */
+  abstract class Range extends DataFlow::Node {
+    /**
+     * Gets a data flow node holding the plaintext version of the email body.
+     */
+    abstract DataFlow::Node getPlainTextBody();
+
+    /**
+     * Gets a data flow node holding the html version of the email body.
+     */
+    abstract DataFlow::Node getHtmlBody();
```

Comment:
This seems to be the only field used in the query. Is there a reason to have and to model the rest?